src/app/widget-config/modal-widget-config/modal-widget-config.component.html

```diff
@@ -168,7 +167,10 @@
                       <mat-option value="capacity">Capacity</mat-option>
                     </mat-select>
                   </mat-form-field>
-
+                  <mat-form-field formGroupName="gauge" class="options-grid-span2">
```

Comment:
Would'nt we be better of with a single new @if statement rather then repeating the template code?

Reply:
The component uses formbuilder to dynamically parse the config object.

The template part is like 7 years old and ugly - ageed! The logic is just to run by the config or form object properties and values and output the approriate field controls. It's all fits in a 2 columns grid layout. It's not super flexible from a UX prespective, but with lots of Widgets and options, it needed something generic. You can use @if and @switch to adjust say, column span base on config properties, if necessary. I'm afraid that by repeating it this way, we will forget to check all config options and conditionnal cases. Months can go by until coding work and we tend to forget these things...plus we have de tests!

I'm all ears to see someone submit a PR with something better.

Reply:
Totally get it.  I've been doing this 30+ years and have my own experience with maintaining "legacy" code, often of my own creation. =)